R/derive_params_growth_height.R

```diff
@@ -0,0 +1,284 @@
+#'   select(USUBJID, VSSTRESN, VSSTRESU, VSDTC) %>%
+#'   rename(
+#'     HEIGHT = VSSTRESN,
+#'     HEIGHTU = VSSTRESU
+#'   )
+#'
+#' advs <- advs %>%
+#'   right_join(., heights, by = c("USUBJID", "VSDTC"))
+#'
+#' advs_under2 <- advs %>%
+#'   filter(AGECUR_M <= 2)
+#'
+#' advs_over2 <- advs %>%
+#'   filter(AGECUR_M > 2)
+#'
+#' who_under2 <- bind_rows(
+#'   (admiralpeds::who_wt_for_lgth_boys %>%
+#'     mutate(
+#'       SEX = "M",
+#'       height_unit = "cm"
+#'     )
+#'   ),
+#'   (admiralpeds::who_wt_for_lgth_girls %>%
+#'     mutate(
+#'       SEX = "F",
+#'       height_unit = "cm"
+#'     )
+#'   )
+#' ) %>%
+#'   rename(
+#'     HEIGHT = Length,
+#'     HEIGHTU = height_unit
+#'   )
+#'
+#' who_over2 <- bind_rows(
+#'   (admiralpeds::who_wt_for_ht_boys %>%
+#'     mutate(
+#'       SEX = "M",
+#'       height_unit = "cm"
+#'     )
+#'   ),
+#'   (admiralpeds::who_wt_for_ht_girls %>%
+#'     mutate(
+#'       SEX = "F",
+#'       height_unit = "cm"
+#'     )
+#'   )
+#' ) %>%
+#'   rename(
+#'     HEIGHT = Height,
+#'     HEIGHTU = height_unit
+#'   )
+#'
+#'
+#' derive_params_growth_height(
+#'   advs_under2,
+#'   sex = SEX,
+#'   height = HEIGHT,
+#'   height_unit = HEIGHTU,
+#'   meta_criteria = who_under2,
+#'   parameter = VSTESTCD == "WEIGHT",
+#'   analysis_var = VSSTRESN,
+#'   set_values_to_sds = exprs(
+#'     PARAMCD = "WTASDS",
+#'     PARAM = "Weight-for-age z-score"
+#'   ),
+#'   set_values_to_pctl = exprs(
+#'     PARAMCD = "WTAPCTL",
+#'     PARAM = "Weight-for-age percentile"
+#'   )
+#' )
+#'
+#' derive_params_growth_height(
+#'   advs_over2,
+#'   sex = SEX,
+#'   height = HEIGHT,
+#'   height_unit = HEIGHTU,
+#'   meta_criteria = who_over2,
+#'   parameter = VSTESTCD == "WEIGHT",
+#'   analysis_var = VSSTRESN,
+#'   set_values_to_sds = exprs(
+#'     PARAMCD = "WTASDS",
+#'     PARAM = "Weight-for-age z-score"
+#'   ),
+#'   set_values_to_pctl = exprs(
+#'     PARAMCD = "WTAPCTL",
+#'     PARAM = "Weight-for-age percentile"
+#'   )
+#' )
+derive_params_growth_height <- function(dataset,
+                                        sex,
+                                        height,
+                                        height_unit,
+                                        meta_criteria,
+                                        parameter,
+                                        analysis_var,
+                                        set_values_to_sds = NULL,
+                                        set_values_to_pctl = NULL) {
+  # Apply assertions to each argument to ensure each object is appropriate class
+  sex <- assert_symbol(enexpr(sex))
+  height <- assert_symbol(enexpr(height))
+  height_unit <- assert_symbol(enexpr(height_unit))
+  analysis_var <- assert_symbol(enexpr(analysis_var))
+  assert_data_frame(dataset, required_vars = expr_c(sex, height, height_unit, analysis_var))
+  assert_data_frame(meta_criteria, required_vars = exprs(SEX, HEIGHT, HEIGHTU, L, M, S))
```

Comment:
@zdz2101 Can you change the input variables names HEIGHT/HEIGHTU coming from meta_criteria? as it's confusing when length will get selected, indeed in the [ADVS peds template](https://phuseaccount.sharepoint.com/:w:/r/sites/Pharmaverse-admiral/_layouts/15/Doc.aspx?sourcedoc=%7B5F12E4C8-0DB7-487E-B155-7C28D81B54D0%7D&file=ADVS_peds_template_draft.docx&action=default&mobileredirect=true), it should renamed as HEIGHT_LENGTH/HEIGHT_LENGTHU.